common/core/src/main/java/com/vmware/connectors/common/payloads/response/Card.java

```diff
@@ -360,9 +373,34 @@ public Builder addTag(String tag) {
          */
         @SuppressWarnings("PMD.UnnecessaryLocalBeforeReturn")
         public Card build() {
+            card.hash = computeHash();
+
             Card completedCard = this.card;
             reset();
             return completedCard;
         }
+
+        private String computeHash() {
+            String result = "";
+            if (card.name != null) {
+                result += card.name;
+            }
+
+            if (card.header != null) {
+                result += card.header.toString();
```

Comment:
Calculating hash on ```Map``` and ```Set``` objects make ```hash``` value change every time. I have tried including these fields, but ```hash``` value tend to change every time.

Reply:
I tried even using LinkedHashMap and SortedMap implementations. But ```hash``` values tends to change every time.

Reply:
Even if we write ```computeHash``` method inside ```CardHeader```, ```CardAction```, ```CardBody```, finally we will end up converting each field in that class to ```ToString``` and calculating```SHA-1```. 

In this case, if we add any new fields we might miss to include adding the new field in ```hash``` calculation.

When we use ````ToStringBuilder.reflectionToString(this, SHORT_PREFIX_STYLE);```, the new fields are automatically included in the hash computation, since we are calling ```toString``` from the ```computeHash``` method.

Reply:
If your library doesn't have a way to hash Set/Map independent of order, then you will have to sort it so it is consistent (sort the Set by its values, set the Map by its keys).

Reply:
If it's not clear, what I'm saying is: You have to extract the values out of the set/map in a certain order when computing the hash.

You aren't going to be able to just toString the Set/Map (again, we shouldn't be using toString for this computation anyway).

Reply:
Got it.. Will change the code accordingly..!!